src/main/java/uk/gov/hmcts/probate/service/FetchDraftCaseService.java

```diff
@@ -26,76 +30,86 @@ public class FetchDraftCaseService {
     private final ServiceRequestClient serviceRequestClient;
     private final NotificationService notificationService;
     private final CaveatQueryService caveatQueryService;
+    private final ElasticSearchRepository elasticSearchRepository;
+    private static final String DRAFT_CASES_QUERY = "templates/elasticsearch/caseMatching/"
+            + "draft_cases_date_range_query.json";
 
-    public void fetchGORCases(String startDate, String endDate) {
+    public void fetchDraftCases(String startDate, String endDate, CcdCaseType ccdCaseType) {
         try {
-            log.info("Fetch GOR cases upto date: {}", endDate);
-            List<ReturnedCaseDetails> cases = caseQueryService.findDraftCases(startDate, endDate);
-            log.info("Found {} cases with draft state", cases.size());
-
-            List<ReturnedCaseDetails> successfulPaymentCases = new ArrayList<>();
-
-            for (ReturnedCaseDetails returnedCaseDetails : cases) {
-                boolean isPaymentSuccessful = processPayment(returnedCaseDetails.getId().toString());
-
-                if (isPaymentSuccessful) {
-                    log.info("Payment status is Success for case id: {}", returnedCaseDetails.getId());
-                    successfulPaymentCases.add(returnedCaseDetails);
-                } else {
-                    log.info("Payment status is not Success for case id: {}", returnedCaseDetails.getId());
-                }
+            SecurityDTO securityDTO = securityUtils.getUserBySchedulerTokenAndServiceSecurityDTO();
+            String caseTypeName = ccdCaseType.getName().equals(CAVEAT.getName())
```

Comment:
if we care about restricting the use to the two options here (which the current implementation here does not) we could do something like:
```suggestion
            String caseTypeName = ccdCaseType.getName().equals(CAVEAT.getName())
            String caseTypeName = switch (ccdCaseType) {
                case CAVEAT, GRANT_OF_REPRESENTATION -> ccdCaseType.getName();
                default -> throw new IllegalArgumentException(...);
            };
```

or if we don't care: 
```suggestion
            String caseTypeName = ccdCaseType.getName();
```